### PR TITLE
update: use cgo instead of uds

### PR DESCRIPTION
**Pull request by jundizhou**
## Description
probe:
 Sending data using CGO mode
 Update struct of kindlingEvent 
 Remove probe Dockerfile 
 Remove probe compiler's scripts
 Use cmake instead of bazel

collector:
 Receiver data using CGO mode
 Update Dockerfile
 Update compiler scripts

## How Has This Been Tested?
recompiler and test all

for issue #265

### CODE REVIEW

File: collector/receiver/cgoreceiver/cgoreceiver.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		select {
+		case <-r.stopCh:
+			r.shutdownWG.Done()
+			return
+		case <-timer.C:
+			r.telemetry.Logger.Info("Total number events received: ", zap.Int("events", r.eventCount))
+			r.eventCount = 0
+			r.telemetry.Logger.Info("Current channel size: ", zap.Int("channel size", len(r.eventChannel)))
+		}
+	}
```
Comment (NeJan2020) [inline collector/receiver/cgoreceiver/cgoreceiver.go:93]: Was that necessary to print this Metric every second

File: deploy/agent/kindling-deploy.yml [hunk 2]
```diff
@@ -39,21 +33,24 @@ spec:
...
+              fieldPath: spec.nodeName
         readinessProbe:
           exec:
             command:
-            - cat
-            - /opt/kernel-support
-          initialDelaySeconds: 5
-          periodSeconds: 5
+              - cat
+              - /opt/kernel-support
```
Comment (NeJan2020) [inline deploy/agent/kindling-deploy.yml:50]: Why not keep the `initialDelaySeconds` and `periodSeconds`?

File: deploy/scripts/build.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+cd ../../
+mkdir -p probe/build
+cd probe/build
+cmake -DBUILD_DRIVER=OFF -DPROBE_VERSION=0.1.1dev ..
+make
```
Comment (NeJan2020) [inline deploy/scripts/build.sh:5]: It seems we have to set the `PROBE_VERSION` manually.

File: deploy/scripts/build.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+if [ ! -f "$libKindlingPath" ]; then
+  echo "compiler libkindling failed! exit!"
+
+else
+  cp -rf ./src/libkindling.so ../../collector/docker/libso
+  cp -rf ./src/libkindling.so /usr/lib64/
+  cd ../../collector/
+  go build
+  collectorPath="./collector"
+  if [ ! -f "$collectorPath" ]; then
```
[the remark this answers is not in the document]
Comment (NeJan2020) [reply deploy/scripts/build.sh:16]: Using the `collector-version-build.sh`  to replace `go build`

File: deploy/agent/kindling-deploy.yml [hunk 5]
```diff
@@ -16,20 +16,14 @@ spec:
     spec:
       serviceAccount: kindling-agent
       containers:
-      - name: kindling-probe
-        image: kindlingproject/kindling-probe:latest
+      - name: kindling-collector
+        image: registry.cn-hangzhou.aliyuncs.com/zjd-kindling/zjdkindling:new1
         imagePullPolicy: Always
```
Comment (NeJan2020) [inline deploy/agent/kindling-deploy.yml:21]: We should not use a personal image

File: collector/receiver/cgoreceiver/cgoreceiver.go (hunk 1 above)
Comment (dxsup) [reply collector/receiver/cgoreceiver/cgoreceiver.go:93]: No, I will open another PR to introduce opentelemetry to record metrics.

**Review by NeJan2020 — APPROVED** [review]

**Pull request merged**